api/src/main/java/org/pentaho/platform/api/genericfile/IGenericFileService.java

```diff
@@ -19,16 +19,118 @@
 
 import edu.umd.cs.findbugs.annotations.NonNull;
 import edu.umd.cs.findbugs.annotations.Nullable;
+import org.pentaho.platform.api.genericfile.exception.AccessControlException;
+import org.pentaho.platform.api.genericfile.exception.InvalidPathException;
 import org.pentaho.platform.api.genericfile.exception.OperationFailedException;
 import org.pentaho.platform.api.genericfile.model.IGenericFileTree;
 
+/**
+ * The {@code IGenericFileService} interface contains operations to access and modify generic files.
+ *
+ * @see GenericFilePath
+ * @see IGenericFileProvider
+ * @see IGenericFileTree
+ */
 public interface IGenericFileService {
+  /**
+   * Clears the cache of folder trees, for all generic file providers, for the current user session.
+   *
+   * @throws OperationFailedException If the operation fails for some (checked) reason.
+   * @see #getFolderTree(GetTreeOptions)
+   * @see #createFolder(GenericFilePath)
+   */
   void clearFolderCache() throws OperationFailedException;
 
+  /**
+   * Gets a tree of folders.
+   * <p>
+   * The results of this method are cached. To ensure fresh results, the {@link #clearFolderCache()} should be called
+   * beforehand.
+   *
+   * @param options The operation options. These control, for example, whether to return the full tree,
+   *                a subtree of a given base path, as well as the depth of the returned folder tree,
+   *                amongst other options.
+   *                <p>
+   *                When the {@link GetTreeOptions#getBasePath() base path option} is {@code null},
+   *                then the returned tree should be rooted at the single provider's root path, if there's a single
+   *                provider, or at the abstract, top-most root whose children are the providers' root paths, if there
+   *                are multiple providers.
+   *                Otherwise, the returned tree is rooted at the specified base path.
+   * @return The folder tree.
+   * @throws AccessControlException   If the user of the current session does not have permission to browse the
```

Comment:
A number of these javadocs mention exceptions that aren't specified in the method signatures, am I missing/misunderstanding something?